modules/launcher/src/main/java/org/mule/runtime/module/launcher/log4j2/LoggerContextCache.java

```diff
@@ -214,4 +227,32 @@ public void dispose() {
     disposedContexts.invalidateAll();
     disposedContexts.cleanUp();
   }
+
+  /**
+   * Marks a {@link MuleLoggerContext} under construction
+   * @param classLoader {@link ClassLoader} that owns the {@link MuleLoggerContext}
+   * @throws RecursiveLoggerContextInstantiationException if a {@link MuleLoggerContext} is already under construction
+   */
+  private void setLoggerContextUnderConstruction(ClassLoader classLoader) {
+    if (isLoggerContextUnderConstruction()) {
```

Comment:
The verification that a the logger is under construction should be outside of the setter. The setter should only set the value.